src/cardstack/al-cardstack-view.ts

```diff
@@ -66,38 +67,59 @@ export abstract class AlCardstackView< EntityType=any,
         }
         let entities = await this.fetchData( true );
 
-        if( this.localPagination ) {
-            this.rawCards = entities;
-            this.resetPagination(this.rawCards.length);
-            entities = entities.slice( 0, Math.min(this.itemsPerPage, entities.length ));
-        }
-
+        this.rawCards = [];
+        this.filteredCards = [];
         this.cards = [];
-        this.ingest( entities );
 
+        let ingestedCards = this.ingest( entities );
+
+        // if we have pagination enable just load a section of data
+        if( this.localPagination ) {
+            this.rawCards = ingestedCards;
```

Comment:
these 2 lines can go above the if statement, as they are the same

Reply:
thanks and done